src/datasources/postgres/ScheduledEventDataSource.ts

```diff
@@ -162,24 +163,30 @@ export default class PostgreScheduledEventDataSource
   }
 
   async getAll(filter: ScheduledEventFilter): Promise<ScheduledEvent[]> {
-    if (!filter.instrumentIds?.length) {
-      return [];
-    }
-
-    const qb = database<ScheduledEventRecord>(this.tableName)
-      .select()
-      .whereIn('instrument_id', filter.instrumentIds)
-      .orderBy('starts_at');
-
-    if (filter.startsAt && filter.endsAt) {
-      qb.where('starts_at', '<=', filter.endsAt).andWhere(
-        'ends_at',
-        '>=',
-        filter.startsAt
-      );
-    }
-
-    const scheduledEventRecords = await qb;
+    const scheduledEventRecords: ScheduledEventRecord[] =
+      await database<ScheduledEventRecord>(this.tableName)
+        .select()
+        .modify((query) => {
+          if (filter.startsAt && filter.endsAt) {
+            query
+              .where('starts_at', '<=', filter.endsAt)
+              .andWhere('ends_at', '>=', filter.startsAt);
+          }
+
+          if (filter.instrumentIds?.length && filter.localContactIds?.length) {
```

Comment:
For readability less coe and stricter data type checking I suggest adding helper function called `isSet()`

So it looks like
`if (isSet(filter.instrumentIds) && isSet(filter.localContactIds)) {`